lib/Utilities/Timers.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <chrono>
+#include <string>
+#include <Logger/Logger.h>
+
+namespace arangodb { namespace Utilities {
+
+struct timer {
+  using ns = ::std::chrono::nanoseconds;
+  using clock = ::std::chrono::high_resolution_clock;
+  using point = ::std::chrono::time_point<clock,ns>;
+  std::string _name;
+  point _start;
+  bool _released;
+
+  timer(std::string const& name = "unnamed")
+    : _name(" - " + name)
+    , _start(clock::now())
+    , _released(false)
+    {}
+
+  ~timer(){
+    if(!_released){
+      release();
+    }
+  }
+
+  void release(){
+    LOG_TOPIC(ERR, Logger::FIXME) << "## ## ## timer" << _name << ":" << std::fixed << diff(_start, clock::now()) / (double) std::giga::num << "s";
```

Comment:
Should this be on a different level/topic? Maybe it's own `TIMER` topic and `INFO`?

Reply:
This timer should only be used during debugging therefor ERR is really what i want / need.

Reply:
In that case, can you add a class comment specifying that it is only intended for use in debugging and shouldn't make it into any committed code?